src/server/response.h

```diff
@@ -47,7 +47,7 @@ class Response {
 
     static std::unique_ptr<Response> build(const InternalServer& server);
     static std::unique_ptr<Response> build_304(const InternalServer& server, const ETag& etag);
-    static std::unique_ptr<Response> build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName);
+    static std::unique_ptr<Response> build_404(const InternalServer& server, const RequestContext& request, const std::string& bookName, const std::string& details="", const std::string& bookTitle="");
```

Comment:
It makes more sens to have bookTitle just after bookName.
And maybe not make `bookTitle` optional.

Either we have a bookName (and probably a bookTitle) or we don't.
But it is possible to have a bookName/bookTitle without details.

Reply:
> It makes more sens to have bookTitle just after bookName.
I did not do this as 404 is being used on multiple methods and I would have to change all of them to do this. I just wanted a go-ahead from the mentors before I did something like that.